GCP Cloud Spending Analysis
- Hanina Marchia Putri (5027261012)
- Marchelio Fahrul Rahmansyah (5027261027)
- Muhammad Affan Arsyad (5027261081)
Topik: Cloud Computing

Komputasi awan seperti Google Cloud Platform (GCP) memungkinkan pengguna menyewa sumber daya komputasi, penyimpanan, dan jaringan dengan model bayar sesuai pemakaian. Model ini fleksibel, tetapi biayanya sulit diprediksi karena bergantung pada jenis layanan, region, volume penggunaan, dan durasi pemakaian. Sumber daya juga sering tidak dimanfaatkan optimal, misalnya utilisasi CPU dan memori rendah tetapi tetap ditagih penuh. Karena itu, pengelolaan biaya awan perlu didukung analisis berbasis data.
Dataset dalam proyek ini memuat 1.000 catatan penggunaan GCP pada 30 Juni sampai 29 Agustus 2024, mencakup 25 layanan di 12 region. Variabelnya meliputi kuantitas penggunaan, utilisasi CPU dan memori, trafik jaringan, durasi, dan biaya. Melalui eksplorasi data (EDA), proyek ini bertujuan memahami distribusi variabel, menemukan pola dan pencilan, serta melihat faktor yang memengaruhi biaya layanan awan.

1.Region mana yang paling mahal, dan apakah selisihnya bermakna secara statistik (ANOVA atau Kruskal-Wallis)?

2.Region atau layanan mana yang trafiknya paling padat?

3.Berapa persen resource yang under-utilized (CPU dan memori di bawah 20%) tetapi tetap berbiaya besar?

In [23]:
import pandas as pd

df = pd.read_csv(r"C:\Users\Hanina Marchia\Project 1 EDA Statprob\gcp_final_approved_dataset.csv")
print(df.shape)
df.head()

(1000, 15)


,Resource ID,Service Name,Usage Quantity,Usage Unit,Region/Zone,CPU Utilization (%),Memory Utilization (%),Network Inbound Data (Bytes),Network Outbound Data (Bytes),Usage Start Date,Usage End Date,Cost per Quantity ($),Unrounded Cost ($),Rounded Cost ($),Total Cost (INR)
0,res-ST6BAJ2N,Cloud Dataproc,954.9843,Requests,europe-north1,92.72,70.42,77097035547,8.268593e+10,01-08-2024 22:24,07-08-2024 06:54,5.24,5004.12,5004,415332
1,res-TYGNR0RV,Pub/Sub,479.6348,GB,europe-west1,37.78,46.80,8239357017,1.135149e+10,23-08-2024 09:18,25-08-2024 07:26,9.82,4710.01,4710,390930
2,res-S3I9C869,BigQuery,114.4129,GB,southamerica-east1,82.54,59.47,23358691883,2.345081e+10,18-07-2024 11:13,22-07-2024 00:45,7.37,843.22,843,69969
3,res-1RY9BZ6G,Cloud Endpoints,413.1930,GB,us-central1,86.68,97.63,53404385778,6.078019e+10,20-07-2024 00:15,21-07-2024 10:16,4.41,1822.18,1822,151226
4,res-S3HQYIZ2,Cloud Spanner,335.9892,GB,asia-southeast1,82.95,21.58,10795270461,1.416138e+10,21-08-2024 21:25,25-08-2024 00:21,6.09,2046.17,2046,169818


1. Sumber data: https://www.kaggle.com/datasets/sairamn19/gcp-cloud-billing-data, Lisensi: Community Data License Agreement - Sharing - Version 1.0
2. Dataset GCP-Cloud-Billing-Data yang dipublikasikan oleh sairamn19 melalui Kaggle. Dataset berisi data penggunaan dan biaya layanan Google Cloud Platform (GCP). Berdasarkan data yang tersedia, periode penggunaan tercatat dari 30 Juni hingga 29 Agustus 2024. Informasi mengenai metode pengumpulan data dan waktu pengumpulan dataset tidak dijelaskan secara memadai pada halaman sumber yang dapat diverifikasi.
3. Jumlah baris: 1000, jumlah kolom: 15
(1000*15)
4. Kamus Data
| Kolom | Arti | Jenis Variabel | Skala Pengukuran | Satuan | Contoh |
|---|---|---|---|---|---|
| Resource ID | ID unik untuk mengidentifikasi resource GCP | Kualitatif | Nominal | Tidak ada | res-ST6BAJ2N |
| Service Name | Nama layanan Google Cloud yang digunakan | Kualitatif | Nominal | Tidak ada | Cloud Spanner |
| Usage Quantity | Jumlah penggunaan suatu layanan atau resource | Kuantitatif | Rasio | Sesuai Usage Unit | 335.9892 |
| Usage Unit | Satuan yang digunakan untuk mengukur penggunaan resource | Kualitatif | Nominal | Tidak ada | GB |
| Region/Zone | Region atau zona tempat resource digunakan | Kualitatif | Nominal | Tidak ada | asia-southeast1 |
| CPU Utilization | Persentase penggunaan CPU pada resource | Kuantitatif | Rasio | % | 82.95% |
| Memory Utilization | Persentase penggunaan memori pada resource | Kuantitatif | Rasio | % | 21.58% |
| Network Inbound Data | Jumlah data yang masuk ke resource melalui jaringan | Kuantitatif | Rasio | Byte | 77097035547 |
| Network Outbound Data | Jumlah data yang keluar dari resource melalui jaringan | Kuantitatif | Rasio | Byte | 82685930000 |
| Usage Start Date | Tanggal dan waktu mulai penggunaan resource | Kuantitatif | Interval | Tanggal & waktu | 01-08-2024 22:24 |
| Usage End Date | Tanggal dan waktu berakhirnya penggunaan resource | Kuantitatif | Interval | Tanggal & waktu | 07-08-2024 06:54 |
| Cost per Quantity | Biaya untuk setiap satuan penggunaan resource | Kuantitatif | Rasio | USD | 5.24 |
| Unrounded Cost | Total biaya penggunaan sebelum pembulatan | Kuantitatif | Rasio | USD | 5004.12 |
| Rounded Cost | Total biaya penggunaan setelah pembulatan | Kuantitatif | Rasio | USD | 5004 |
| Total Cost | Total biaya penggunaan resource dalam mata uang rupee India | Kuantitatif | Rasio | INR | 415332 |
   

In [10]:
df.info()                 # tipe data setiap kolom

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 15 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Resource ID                    1000 non-null   str    
 1   Service Name                   1000 non-null   str    
 2   Usage Quantity                 1000 non-null   float64
 3   Usage Unit                     1000 non-null   str    
 4   Region/Zone                    1000 non-null   str    
 5   CPU Utilization (%)            1000 non-null   float64
 6   Memory Utilization (%)         1000 non-null   float64
 7   Network Inbound Data (Bytes)   1000 non-null   int64  
 8   Network Outbound Data (Bytes)  1000 non-null   float64
 9   Usage Start Date               1000 non-null   str    
 10  Usage End Date                 1000 non-null   str    
 11  Cost per Quantity ($)          1000 non-null   float64
 12  Unrounded Cost ($)             1000 non-null   float64
 13  

In [9]:
df.isna().sum()           # jumlah data kosong per kolom

Resource ID                      0
Service Name                     0
Usage Quantity                   0
Usage Unit                       0
Region/Zone                      0
CPU Utilization (%)              0
Memory Utilization (%)           0
Network Inbound Data (Bytes)     0
Network Outbound Data (Bytes)    0
Usage Start Date                 0
Usage End Date                   0
Cost per Quantity ($)            0
Unrounded Cost ($)               0
Rounded Cost ($)                 0
Total Cost (INR)                 0
dtype: int64

In [24]:
df.duplicated().sum()     # jumlah baris duplikat

np.int64(0)

In [22]:
print('Variabel Numerik')
hasil = pd.DataFrame({
    "Usage Quantity": [
        df["Usage Quantity"].mean(),
        df["Usage Quantity"].median(),
        df["Usage Quantity"].mode()[0],
        df["Usage Quantity"].var(),
        df["Usage Quantity"].std(),
        df["Usage Quantity"].quantile(0.25),
        df["Usage Quantity"].quantile(0.75)
    ],
    
    "CPU Utilization (%)": [
        df["CPU Utilization (%)"].mean(),
        df["CPU Utilization (%)"].median(),
        df["CPU Utilization (%)"].mode()[0],
        df["CPU Utilization (%)"].var(),
        df["CPU Utilization (%)"].std(),
        df["CPU Utilization (%)"].quantile(0.25),
        df["CPU Utilization (%)"].quantile(0.75)
    ],
    
    "Total Cost (INR)": [
        df["Total Cost (INR)"].mean(),
        df["Total Cost (INR)"].median(),
        df["Total Cost (INR)"].mode()[0],
        df["Total Cost (INR)"].var(),
        df["Total Cost (INR)"].std(),
        df["Total Cost (INR)"].quantile(0.25),
        df["Total Cost (INR)"].quantile(0.75)
    ]
}, index=[
    "Mean",
    "Median",
    "Mode",
    "Variance",
    "Standard Deviation",
    "Q1",
    "Q3"
])

hasil #Nilai plus: hitung varians atau simpangan baku untuk satu kolom secara manual dengan rumus (misalnya memakai 10 baris pertama), lalu cocokkan dengan hasil pandas. Perhatikan bahwa var() dan std() di pandas memakai pembagi n − 1.

Variabel Numerik


,Usage Quantity,CPU Utilization (%),Total Cost (INR)
Mean,383.343530,50.943680,1.772228e+05
Median,336.721800,52.030000,1.038330e+05
Mode,91.866700,0.690000,2.631100e+04
Variance,89889.431411,830.865741,2.982042e+10
Standard Deviation,299.815662,28.824742,1.726859e+05
Q1,106.818150,26.740000,4.154150e+04
Q3,616.048825,75.647500,2.703725e+05
